packages/docgen/lib/get-intermediate-representation.js

```diff
@@ -20,7 +20,7 @@ const hasVariableWithName = ( node, name ) =>
 	node.declarations.some( ( declaration ) => {
 		if ( declaration.id.type === 'ObjectPattern' ) {
 			return declaration.id.properties.some(
-				( property ) => property.key.name === name
+				( property ) => property.key && property.key.name === name
```

Comment:
If you want to catch both `a` and `b` in a `const { a, ...b } = ...` declaration, you need to check:
```js
property.key?.name === name || property.argument?.name === name
```
The first one is for normal keys like `a`, the second one for rest keys like `b`. Here is what ASTExplorer shows for `const { a, ...b }` syntax:

<img width="532" alt="Screenshot 2024-04-16 at 10 59 35" src="https://github.com/WordPress/gutenberg/assets/664258/f34222ab-7cce-4985-b8e7-d46bf72fddc5">

Reply:
I'm not entirely sure what this function does to be honest, but I can make that change :P

Reply:
That was interesting!

Reply:
I think it's used to find a connection between an exported name and a JSDoc comment. For example:
```js
/**
 * Some comment with documentation about `b`.
 */
const { a, ...b } = foo();

export { b };
```
Here the comment will be used as a documentation for the exported name `b`.

Reply:
Pushed a test for this https://github.com/WordPress/gutenberg/pull/60748/commits/2c7526dbe494bedbc52ac023b510bf02138f6ee4 but Jarda beat me to the suggestion :)

Reply:
I've got another test to add, I've disabled auto-merge. I'll merge after.